aws/resource_aws_waf_regex_match_set.go

```diff
@@ -0,0 +1,176 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegexMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegexMatchSetCreate,
+		Read:   resourceAwsWafRegexMatchSetRead,
+		Update: resourceAwsWafRegexMatchSetUpdate,
+		Delete: resourceAwsWafRegexMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"regex_match_tuple": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      resourceAwsWafRegexMatchSetTupleHash,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"field_to_match": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data": {
+										Type:     schema.TypeString,
+										Optional: true,
+										StateFunc: func(v interface{}) string {
+											value := v.(string)
+											return strings.ToLower(value)
+										},
```

Comment:
one-liner? I'm not convinced either way it's good style but I'll suggest it and let you decide:
```
return strings.ToLower(v.(string))
```